concent_api/core/subtask_helpers.py

```diff
@@ -103,15 +111,18 @@ def _update_timed_out_subtask(subtask: Subtask) -> None:
         )
     elif subtask.state == Subtask.SubtaskState.ADDITIONAL_VERIFICATION.name:  # pylint: disable=no-member
         task_to_compute = deserialize_message(subtask.task_to_compute.data.tobytes())
-        # Worker makes a payment from requestor's deposit just like in the forced acceptance use case.
-        bankster.finalize_payment(
-            subtask_id=task_to_compute.subtask_id,
-            concent_use_case=ConcentUseCase.FORCED_ACCEPTANCE,
-            requestor_ethereum_address=task_to_compute.requestor_ethereum_address,
-            provider_ethereum_address=task_to_compute.provider_ethereum_address,
-            subtask_cost=task_to_compute.price,
+        finalize_deposit_claim(
+            subtask_id=subtask.subtask_id,
+            concent_use_case=ConcentUseCase.ADDITIONAL_VERIFICATION,
+            ethereum_address=task_to_compute.requestor_ethereum_address,
```

Comment:
Isn't `_update_timed_out_subtask()` usually called inside a transaction? If so, `finalize_payment()` will not be able to commit its own transactions.

BTW, I think we should make it easier to detect situations like this automatically. We should have a wrapper over `transaction.atomic()` that fails if a transaction is already in progress and use it everywhere. We need to detect and eliminate any nested transactions.

Reply:
:+1: done